ribs/schedulers/_bandit_scheduler.py

```diff
@@ -240,13 +240,18 @@ def ask(self):
 
             self._restarts = emitter_restarts
         else:
-            # Reselect all emitters.
+            # Reselect all active emitters.
             reselect = self._active_arr.copy()
 
-        # If no emitters are active, activate the first num_active.
-        if not self._active_arr.any():
+        # If not enough emitters are active, activate the first num_active.
+        diff = self._num_active - self._active_arr.sum()
```

Comment:
Could you make these names more descriptive, like `num_needed` and `active_i`?

Reply:
I don't understand where you expect `active_i`.
No problem for `num_needed\, you're right it's more explicit.

Reply:
Oh I meant that `i` would be renamed to `active_i`. Makes more sense to me since `i` is a common name and it's going to be available after the loop is done, so it would be easy for it to get confused in later pieces of code.